spec/integration/stats_partition_generator_job_spec.rb

```diff
@@ -0,0 +1,38 @@
+require_relative '../spec_helper'
+
+RSpec.describe ThreeScale::Backend::Stats::PartitionGeneratorJob do
+  let(:service_id) { '123456' }
+  let(:applications) { %w[1] }
+  let(:metrics) { %w[10] }
+  let(:users) { %w[100] }
+  let(:from) { Time.new(2002, 10, 31).to_i }
+  let(:to) { Time.new(2002, 11, 30).to_i }
+  let(:job) do
+    ThreeScale::Backend::Stats::DeleteJobDef.new(
+      service_id: service_id,
+      applications: applications,
+      metrics: metrics,
+      users: users,
+      from: from,
+      to: to
+    )
+  end
+  let(:stats_queue) { :stats }
+  let(:configuration) { ThreeScale::Backend.configuration }
+  let(:num_keys_generated) { ThreeScale::Backend::Stats::KeyGenerator.new(job).keys.count }
+
+  before :each do
+    ThreeScale::Backend::Worker::QUEUES.each { |queue| Resque.remove_queue(queue) }
+    configuration.stats.delete_partition_batch_size = 100
```

Comment:
Is the configuration reset on every test? I don't remember right now, but I don't see anything in the test helpers.

If this is not reset, we should add an `after` clause to restore the original value or add a reset of the config to the global `before`.

Reply:
I think when testing, each test has to `set` its own values.  

assuming a certain configuration key X will have value V is an error and goes agains test isolation paradigm

Reply:
I agree with that, but in this case, we set the config with these values in a test helper: https://github.com/3scale/apisonator/blob/master/test/test_helpers/configuration.rb

I think it's reasonable to expect that those defaults apply to each test. However, in order to ensure that each test needs to restore the config to its original values if they modify them, or we need to do that in a global `before`.

You don't have to fix this now because it's a more general problem. I see that we have more tests with the same case, but I think we should fix this at some point.

Reply:
that test_helper is for Test::Unit. We are discussing a RSpec test

Reply:
But it's used in RSpec too https://github.com/3scale/apisonator/blob/bd63f9d59476d7913fa662ceebc33625dc07a982/spec/spec_helper.rb#L37

It's confusing. We should probably separate the helpers so it's clear that they're used from both test suites.